### DISCUSSION TITLE
Internal testing of production site

### DISCUSSION THREAD

**Discussion by TorrentofShame** [KnightHacks/hackathon-2021-frontend#51] [General]
# Send your feedback here for your experience of using [https://2021.knighthacks.org/](https://2021.knighthacks.org/)

**Comment by APherwani** [comment]
We need some more info on the beginner and advanced tracks in the about us section

**Reply by ElijahMSmith** [reply]
There definitely needs to be some level of explanation about what the tracks are for, what does it mean to follow one track or the other, and what kind of restrictions that puts on you (if any). I don't actually know what the tracks are so I can't comment any suggestions.

**Reply by ElbertTse** [reply]
Going to tag onto this but maybe revising the second track to just "intermediate" or "advanced" and explaining that instead.

**Reply by ailpon** [reply]
will be implemented in #52

**Reply by TorrentofShame** [reply]
Instead of talking about the tracks in the about us section, (because let's face it, most people will go directly to the form and ignore the rest of the site) it would be ideal to have some help text appear next to the field when you hover over something that says something like, "what's this?".

That way you don't break ux by forcing them to navigate to another page, reas the stuff there, and go back.

**Comment by ElbertTse** [comment]
It's a bit hard to find the fields that are invalid. One thing that could help is making the text for the error message a bit lighter so it pops out more against the background. I would also suggest highlighting the fields that are invalid or move the invalid message above the field that is invalid but im pretty sure this can be debated.

**Reply by ailpon** [reply]
good points. will be implemented in #53

**Comment by ElbertTse** [comment]
For the Resume, I'd suggest adding a file size limit since it looks like I can submit an 800 mb pdf. Also like it was mentioned earlier, a loading animation or progress bar when submitting. What might be a good idea is to have the resume submission happen earlier in the form and upload it then, so that while its being uploaded, people can fill out the rest of the form.

**Reply by ElbertTse** [reply]
Okay thinking about it now, uploading the resume before submitting a name is probably going to be a bad idea so ignore that. But definitely add some text saying what the upload size limit is and checking that users aren't uploading anything larger

**Reply by TorrentofShame** [reply]
Uploading the resume before creating a hacker is a perfectly reasonable plan.

**Reply by ailpon** [reply]
I think I prefer the idea suggested by you & Simon of uploading the resume before creating the hacker, we just have to think about the best ux for that. telling people their resume files are too big / providing a file limit may result them not really knowing what to do w that info. resume upload will be resolved in #54 & loading animation will be resolved in #55

**Reply by TorrentofShame** [reply]
There currently is no server-side filesize limit simply because we don't really have a limit per document, but I can look into it.

**Reply by APherwani** [reply]
we could probably take care of this on the front end right?
https://gist.github.com/zensh/4975495
This is kind of hacky but it seems promising @aileenpongnon

**Comment by ElijahMSmith** [comment]
I don't exactly like that the popup for having one or more invalid fields is right on the bottom left of the screen. Seems like it should be more visible when it pops up. Is there a reason why it isn't centered?

**Reply by ailpon** [reply]
no reason! that position was decided on my one of the FE teammates. i adjusted it to be centered, like this view better?

@ElijahMSmith https://625d94f5.hackathon-2021-frontend.pages.dev/

**Reply by ElijahMSmith** [reply]
I do like this better. It probably doesn't matter on a small screen for findability, but it looks more natural and on my monitor it's much easier to locate immediately upon failure.

**Reply by ailpon** [reply]
sounds good! it matters a lot, UI is always important. we'll implement touch validation and have this as an added validation layer with the centered design

**Comment by ElijahMSmith** [comment]
Also, the text on the FAQs needs to be tightened up a bit. There are a few formatting things I noticed that aren't super consistent. Of note, there is at least one instance of using the oxford comma and one where you don't that makes the reading a bit off. Another thing would be when talking about what kind of workshops you ran. IMO the capitalization is a bit weird if you don't also put them in quotes, but that could just be me.

**Reply by ailpon** [reply]
will be resolved in #56

**Comment by ElijahMSmith** [comment]
Is there a reason why the required fields aren't marked as required until you try to submit the first time? Seems like it'll cause more people to have to go back and re-fill out parts of the form or to fill in things that aren't required if they think the whole form is.

**Reply by ailpon** [reply]
will be resolved in #53

**Comment by ElijahMSmith** [comment]
I think the lack of text in the about us page is a bit strange. Some information about how the hackathon is being organized, some of the team members or organizers, information about Knight Hacks on whole, things like that.

**Reply by ailpon** [reply]
will be resolved in #52

**Comment by ElijahMSmith** [comment]
Last thing, personally I don't like how the navigation menu animates. The blue box that always starts on the home page, regardless of which page you're actually on, could be done away with or moved to the current page if it's supposed to be an actual indicator. I also think the animation on the pages growing bigger/smaller should be sped up considerably so that the resizing looks a bit smoother.

**Reply by ElbertTse** [reply]
Agree with speeding up the animations

**Reply by ailpon** [reply]
will be resolved in #57

**Comment by TorrentofShame** [comment]
## Accessibility issues

- Hard to see if you are focused on front page buttons and drop-down fields (when using keyboard navigation).
- Cannot access navigation menu nor mute/unmute using tab navigation.
- Cannot access faq items using tab navigation.
- Social icons don't have `alt` text.
- Upload Resume button doesn't meet WCAG fg/bg contrast specs.

## Misc issues
- No `robots.txt`, having one would be good for SEO.
- User is forced to download a massive mp3, should either be streamed or split into smaller chunks.
- No CSP is setup

I recommend that you run the site through Chrome DevTools Lighthouse.

**Reply by ailpon** [reply]
thank you for this
resume button now meeting WCAG standards & social icons have `aria-labels`
tab navigation on homepage will be resolved in #7 
tab navigation on FAQ will be resolved in #56
tab navigation on register will be resolved in #58

CSP setup,`robots.txt`, mp3 refactor will be discussed at next fe sprint meeting

**Reply by ailpon** [reply]
navigation on homepage is majority resolved & navigation on register page is entirely resolved. here's the cloudflare deploy https://3c9448a2.hackathon-2021-frontend.pages.dev including changes from PR #70 and PR #71 . having trouble identifying our volume & menu icons for focus tabbing but I'm thinking of merging the main buttons in first & completing our required MLH work to spend more time on that
